core/src/test/java/org/apache/iceberg/actions/TestSortStrategy.java

```diff
@@ -86,9 +86,6 @@ private List<FileScanTask> tasksForSortOrder(int sortOrderId, int... fileSizesMB
 
   @Test
   public void testInvalidSortOrder() {
-    AssertHelpers.assertThrows("Should not allow an unsorted Sort order", IllegalArgumentException.class,
-        () -> defaultSort().sortOrder(SortOrder.unsorted()).options(Collections.emptyMap()));
```

Comment:
This test fails now as the table has a partition spec:
```
[
  1000: data_bucket: bucket[16](2)
]
```

Resulting in the overall Sort Order to be:

```
[
  bucket[16](2) ASC NULLS FIRST
]
```

Hence, no scope of `IllegalArgumentException`